x/babylon/keeper/msg_server.go

```diff
@@ -14,3 +19,20 @@ type msgServer struct {
 func NewMsgServer(k *Keeper) *msgServer {
 	return &msgServer{k: k}
 }
+
+// UpdateParams updates the params.
+func (ms msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
```

Comment:
Nit: Shouldn't this belong to a different / independent PR?

Or this is needed to get the contracts addresses?

Reply:
It is necessary as eventually one might need to trigger gov prop to call this message in order to set up the Babylon contract addresses. It's not a big change -- most things about this message handler are copied from Babylon e.g., https://github.com/babylonchain/babylon/blob/2a7c91d6f1649912f14667d3a3ee6a2c1792c3d9/x/btccheckpoint/keeper/msg_server.go#L97-L114